controllers/vspheremachine_controller.go

```diff
@@ -274,7 +269,37 @@ func (r machineReconciler) reconcileDeleteVMPre7(ctx *context.MachineContext) er
 	return nil
 }
 
+func (r machineReconciler) findVMPre7(ctx *context.MachineContext) (*infrav1.VSphereVM, error) {
+	// Get ready to find the associated VSphereVM resource.
+	vm := &infrav1.VSphereVM{}
+	vmKey := apitypes.NamespacedName{
+		Namespace: ctx.VSphereMachine.Namespace,
+		Name:      ctx.Machine.Name,
+	}
+	// Attempt to find the associated VSphereVM resource.
+	if err := ctx.Client.Get(ctx, vmKey, vm); err != nil {
+		// If an error occurs finding the VSphereVM resource other than
+		// IsNotFound, then return the error. Otherwise it means the VSphereVM
+		// is already deleted, and that's okay.
+		if !apierrors.IsNotFound(err) {
+			return vm, errors.Wrapf(err, "failed to get VSphereVM %s", vmKey)
+		}
+	}
+	return vm, nil
+}
+
 func (r machineReconciler) reconcileNormal(ctx *context.MachineContext) (reconcile.Result, error) {
+	vsphereVM, err := r.findVMPre7(ctx)
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
```

Comment:
`findVMPre7` will never return the IsNotFound error but we still check it here, is that expected?

Reply:
+1. This is related to my comment about the error being wrapped.